### PR TITLE
Create user model class

### PR DESCRIPTION
**Pull request by vicayala82**
## Summary
-Created User model class
    -Created methods for add and remove book of collection
    -Created model relationship user whit book
    -Created BookAlreadyOwnedException class
## Trello Card
https://trello.com/c/2fqH75mu/19-crear-modelo-de-user

### CODE REVIEW

File: training/src/main/java/wolox/training/models/User.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity (name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (juandbc) [inline training/src/main/java/wolox/training/models/User.java]:
```suggestion
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
```

File: training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package wolox.training.exceptions;
+
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.BAD_REQUEST, reason = "Actor Not Found")
```
Comment (juandbc) [inline training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java]: This should return a conflict a status

File: training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    public BookAlreadyOwnedException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public BookAlreadyOwnedException(Throwable cause) {
+        super(cause);
+    }
+
+    public BookAlreadyOwnedException(String message) {
+        super(message);
```
Comment (juandbc) [inline training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java]: Like the other exceptions, you can leave the message here

File: training/src/main/java/wolox/training/models/User.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    @Column(nullable = false)
+    List<Book> books = new ArrayList<>();
+
+    public long getId() {
+        return id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
```
Comment (juandbc) [inline training/src/main/java/wolox/training/models/User.java]:
```suggestion
```

File: training/src/main/java/wolox/training/models/User.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    @Column(nullable = false)
+    private String username;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private LocalDate birthdate;
+
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL)
```
Comment (juandbc) [inline training/src/main/java/wolox/training/models/User.java]: In the most scenarios, it is better to use the merge cascade. Thus, you avoid to delete data and relationships
```suggestion
    @OneToMany(mappedBy = "user", cascade = CascadeType.MERGE)
```

**Review by juandbc — CHANGES_REQUESTED** [review]

File: training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java (hunk 3 above)
Comment (vicayala82) [reply training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java]: I fixed it new commit

**Review by juandbc — APPROVED** [review]

**Pull request merged**